### PR TITLE
Sky Area + Ellipticities

### PR DESCRIPTION
**Pull request by dkorytov**
Also changed positionAngle to be in degrees.

### CODE REVIEW

**Comment by dkorytov** [comment]
Okay, I've updated the version checking to use StrictVersion

File: GCRCatalogs/alphaq.py [hunk 1]
```diff
@@ -23,25 +23,35 @@ def _subclass_init(self, filename, **kwargs):
...
         self.lightcone = kwargs.get('lightcone')
 
+
         with h5py.File(self._file, 'r') as fh:
             self.cosmology = FlatLambdaCDM(
                 H0=fh['metaData/simulationParameters/H_0'].value,
                 Om0=fh['metaData/simulationParameters/Omega_matter'].value,
                 Ob0=fh['metaData/simulationParameters/Omega_b'].value,
             )
             try:
```
Comment (yymao) [inline GCRCatalogs/alphaq.py]: Change Lines 33-50 
```python
    catalog_version = list()
    for version_label in ('Major', 'Minor', 'MinorMinor'):
        try:
            catalog_version.append(fh['metaData/version' + version_label].value)
        except KeyError:
            break
    if not version:
        catalog_version = [2, 0]
    catalog_version = StrictVersion('.'.join(map(str, version)))

config_version = StrictVersion(kwargs.get('version', '0.0'))
```

File: GCRCatalogs/alphaq.py [hunk 2]
```diff
@@ -70,14 +76,23 @@ def _subclass_init(self, filename, **kwargs):
             'velocity_z':         'vz',
         }
 
+        if(catalog_version < StrictVersion('2.1.1')):
```
Comment (yymao) [inline GCRCatalogs/alphaq.py]: Remove unnecessary parentheses 
`if catalog_version < StrictVersion('2.1.1'):`

File: GCRCatalogs/alphaq.py [hunk 3]
```diff
@@ -70,14 +76,23 @@ def _subclass_init(self, filename, **kwargs):
...
+            self._quantity_modifiers.update({
+                'disk_sersic_index':  'diskSersicIndex',
+                'bulge_sersic_index': 'spheroidSersicIndex',
+            })
+            del self._quantity_modifiers['ellipticity_1']
+            del self._quantity_modifiers['ellipticity_2']
+
         if catalog_version == '2.0': # to be backward compatible
```
Comment (yymao) [inline GCRCatalogs/alphaq.py]: Use `StrictVersion` for comparison
`if catalog_version <= StrictVersion('2.0'):`

Comment (dkorytov) [reply GCRCatalogs/alphaq.py]: Okay, it's in there.

File: GCRCatalogs/alphaq.py [hunk 4]
```diff
@@ -116,39 +131,33 @@ def native_quantity_getter(native_quantity):
 
 
     def _get_native_quantity_info_dict(self, quantity, default=None):
```
Comment (yymao) [inline GCRCatalogs/alphaq.py]: Lines 133--160. Try the following
```python

    def _get_native_quantity_info_dict(self, quantity, default=None):
        with h5py.File(self._file, 'r') as fh:
            quantity_key = 'galaxyProperties/' + quantity
            if quantity_key not in fh:
                return default
            modifier = lambda k, v: None if k=='description' and v==b'None given' else v.decode()
            return {k: modifier(k, v) for k, v in fh[quantity_key].attrs.items()}
            

    def _get_quantity_info_dict(self, quantity, default=None):
        q_mod = self.get_quantity_modifier(quantity)
        if callable(q_mod) or (isinstance(q_mod, (tuple, list)) and len(q_mod) > 1 and callable(q_mod[0])):
            warnings.warn('This value is composed of a function on native quantities. So we have no idea what the units are')
            return default
        return self._get_native_quantity_info_dict(q_mod or quantity, default=default)
```

You also need to `import warnings`

**Comment by dkorytov** [comment]
The changed functions work. :)

**Pull request merged**